---
title: "epimutacions: a Bioconductor package to identify outliers in rare diseases DNA methylation data"
author:
  - Leire Abarrategui 
  - Carlos Ruiz-Arenas
  - Carles Hernandez-Ferrer
  - Patricia Ryser-Welch
  - Juan R. Gonzalez
date: "`r Sys.Date()`"
header-includes:
  - \newcommand{\beginsupplement}{\setcounter{table}{0}  
  - \renewcommand{\thetable}{S\arabic{table}} \setcounter{figure}{0}
  - \renewcommand{\thefigure}{S\arabic{figure}}}
  - \renewcommand{\and}{\\}
  - \usepackage{floatrow}
  - \floatsetup[figure]{capposition=top}

abstract: "Epimutations are rare alterations in the methylation pattern at specific loci. Have been demonstrated that epimutations could be the causative factor of some genetic diseases. Nonetheless, no standard methods are available to detect and quantify these alterations.  We have developed `epimutacions` package which implements 6 approaches to identify epimutations in Illumina DNA methylation microarrays. The document describes, package installation; data loading and preprocessing;  epimutation identification, annotation and visualization; and GEO datasets results."

output:
  pdf_document: 
    latex_engine: xelatex
    fig_caption: true
    number_sections: true

    
subtitle: "Supplementary material" 
vignette: >
  %\VignetteIndexEntry{Detection of epimutations with state of the art methods in methylation data} 
  %\VignetteEngine{knitr::rmarkdown} 
  %\VignetteEncoding{UTF-8}
bibliography: refs.bib
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(comment="", warning = FALSE, message = FALSE, cache = TRUE)
```
\newpage
\tableofcontents
\newpage

\beginsupplement 

# Introduction

We created `epimutacions` R/Biconductor package. 
It implements 6 outlier detection statistical methods to identify epimutations 
in Illumina DNA methylation microarray data. `Quantile` [@garg2020survey] and `beta` methods 
test outliers for every genomic location and then group the outlier CpGs into regions. 
`Manova` [@friedrich2017package], `mlm` [@martin2020multivariate], `iForest` [@cortes2021package] and `mahdist` [@maechler2021package] test if the differentially methylated regions identified by bumphunter [@jaffe2012bump] are outliers. The `Quantile` [@barbosa2018identification; @garg2020survey] and `Manova` [@aref2019] methods were previously described in the literature. 

We used two different strategies to assess the performance of our methods and compare them with the methods available in the `ramr` package. We explored the effect of sample size on the identification of 4 epimutations experimentally validated by Garg and colleagues. We calculated the epimutations per individual and epimutations overlap between the methods. 
The epimutacions package allows raw microarray intensities and preprocessed Illumina methylation arrays. To convert the raw microarray intensities into beta values we can call the 6 normalization methods implemented in the minfi package (epi_preprocess). The package provides tools for the epimutations annotation and visualization. 

The name of the package is `epimutacions` (pronounced `É› pi mu ta 'sj ons`) which means epimutations in Catalan, a language from the northeast of Spain.




```{r implementation, echo=FALSE, fig.cap= "Implementation of each outlier detection method", out.width = '90%', fig.align='center'}
knitr::include_graphics(file.path(getwd(),"fig/implementation.png"))
```


# Setup

## Installing the package

The `epimutacions` package can be installed using the following code: 

```{r eval = FALSE}
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("epimutacions")
```


## Loading libraries

```{r, message=FALSE}
library(epimutacions)
```


# Quick start

The workflow in figure \ref{fig:workflow} explains the main analysis in the `epimutacions` package. 

The package allows two different types of inputs: 

 * (1) Case samples `IDAT` files (raw microarray intensities) together with `RGChannelSet` object as reference panel. The reference panel can be supplied by the user or can be selected through the example datasets that the package provides (section \ref{datasets}). 
 
* (2) `GenomicRatioSet` object containing case and control samples.  


The normalization (`epi_preprocess`) converts the raw microarray intensities into usable methylation measurement ($\beta$ values at CpG locus level). As a result, we obtain a  `GenomicRatioSet` object, which can be used as `epimutations()` function input.  The data should contain information about \beta values of CpG sites, phenotype and feature data.   

```{r workflow, echo=FALSE, fig.cap="Allowed data formats, normalization and input types", out.width = '90%', fig.align='center'}
knitr::include_graphics(file.path(getwd(),"fig/workflow.png"))
```


# Datasets {#datasets}

The package contains 3 example datasets adapted from [Gene Expression Omnibus (GEO)](https://www.ncbi.nlm.nih.gov/geo/): 

* (1) 4 case samples IDAT files [(GEO: GSE131350)](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE131350)
* (2) `reference_panel`:  a `RGChannelSet` class object containing 22 healthy individuals [(GEO: GSE127824)](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE12782)
* (3) `methy`: a `GenomicRatioSet` object which includes 49 controls [(GEO: GSE104812)](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE104812) and 3 cases [(GEO: GSE97362)](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi).

We also  included a dataset specifying the 40,408 candidate regions in Illumina 450K array  which could be epimutations (see section \ref{candreg}). 

We created the `epimutacionsData` package in `ExperimentHub`. It contains the reference panel, methy and the candidate epimutations datasets. The package includes the IDAT files as external data.  To access the datasets  we need to install the packages by running the following commands: 

```{r eval = FALSE}
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("ExperimentHub")
```

```{r eval = FALSE}
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("epimutacionsData")
```

Then, we need to load the package and create an `ExperimentHub` object: 

```{r message = FALSE}
library(ExperimentHub)
eh <- ExperimentHub()
query(eh, c("epimutacionsData"))
```


## IDAT files and reference panel 

IDAT files directory in `epimutacionsData` package:

```{r IDAT files}
baseDir <- system.file("extdata", package = "epimutacionsData")
```

The reference panel can be found in `EH6691` record of the  `eh` object: 

```{r message = FALSE}
reference_panel <- eh[["EH6691"]]
```

## Methy {#methy}

The `methy` is stored in `EH6690` record: 


```{r message = FALSE}
methy <- eh[["EH6690"]]
```



## Candidate regions {#candreg}

In Illumina 450K array, probes are unequally distributed along the genome, limiting the number of regions that can fulfil the requirements to be considered an epimutation.  Thus, we have computed a dataset containing all the regions that could be candidates to become an epimutation.


We used the clustering approach from bumphunter to define the candidate epimutations.  We defined a primary dataset with all the CpGs from the Illumina 450K array.  Then, we run bumphunter and selected those regions with at least 3 CpGs and a maximum distance of 1kb between them. As a result, we found 40,408 candidate epimutations. The function `epimutation()`  filters the identified epimutations using these candidate regions. 

The following is the code used to identify the candidate epimutations in Illumina 450K array: 

```{r eval = FALSE}
library(minfi)
# Regions 450K
library(IlluminaHumanMethylation450kanno.ilmn12.hg19)
data(Locations)

### Select CpGs (names starting by cg) in autosomic chromosomes
locs.450 <- subset(Locations, 
                   grepl("^cg", rownames(Locations)) & chr %in% paste0("chr", 1:22))
locs.450GR <- makeGRangesFromDataFrame(locs.450, start.field = "pos", 
                                       end.field = "pos", strand = "*")
locs.450GR <- sort(locs.450GR)
mat <- matrix(0, nrow = length(locs.450GR), ncol = 2, 
              dimnames = list(names(locs.450GR), c("A", "B")))

## Set sample B to all 1
mat[, 2] <- 1

## Define model matrix
pheno <- data.frame(var = c(0, 1))
model <- model.matrix(~ var, pheno)

## Run bumphunter
bumps <- bumphunter(mat, design = model, pos = start(locs.450GR), 
                    chr = as.character(seqnames(locs.450GR)),
                    cutoff = 0.05)$table
bumps.fil <- subset(bumps, L >= 3)
```


The candidate regions  can be found  in `EH6692` record of the  `eh` object: 


```{r message = FALSE, eval = FALSE}
candRegsGR <- eh[["EH6692"]]
```


# Preprocessing

The  `epi_preprocess` function  allows calling the 6 preprocessing methods from `minfi` package [@aryee2014minfi]:

   
+---------------+-----------------------------+----------------------------------------------------------------------+
| Method        |  Function                   | Description                                                          |
+===============+=============================+======================================================================+
| `raw`         | `preprocessRaw`             | Converts the Red/Green channel for an Illumina methylation           |  
|               |                             |  array into methylation signal. This method does not normalize       |
|               |                             |  the data.                                                           |
+---------------+-----------------------------+----------------------------------------------------------------------+
| `illumina  `  | `preprocessIllumina`        |  Implements preprocessing for Illumina methylation                   |
|               |                             |      microarrays as used in Genome Studio.                           |
+---------------+-----------------------------+----------------------------------------------------------------------+
|`swan`         | `preprocessSWAN `           |Subset-quantile Within Array Normalisation (SWAN). It allows Infinium |
|               |                             | I and II type probes on a single array to be normalized together.    |
+---------------+-----------------------------+----------------------------------------------------------------------+
|  `quantile`   |`preprocessQuantile`         |Implements stratified quantile normalization preprocessing for        |                
|               |                             |Illumina methylation microarrays.                                     |
+---------------+-----------------------------+----------------------------------------------------------------------+
| `noob`        | `preprocessNoob`            | Noob (normal-exponential out-of-band) is a background correction     | 
|               |                             | method with dye-bias normalization for                               |
|               |                             | Illumina Infinium methylation arrays.                                |
+---------------+-----------------------------+----------------------------------------------------------------------+
| `funnorm`     | `preprocessFunnorm `        | Functional normalization (FunNorm) is a between-array                |
|               |                             | normalization method for the Illumina Infinium                       |
|               |                             | HumanMethylation450 platform.                                        | 
+---------------+-----------------------------+----------------------------------------------------------------------+

Table: Preprocessing methods description
\label{tab:prep_des}

Each normalization approach has some unique parameters (table \ref{tab:prep_unique_parameter}) which can be modified through `norm_parameters()` function: 

Table: Preprocessing methods unique parameters
\label{tab:prep_unique_parameter}

+----------+-----------------------+-----------------------------------------------------------------------------+
| Method   |Parameters             |Description                                                                  |
+==========+=======================+=============================================================================+
|`illumina`|`bg.correct`\          |Performs background correction\                                              |
|          |`normalize`\           |Performs controls normalization\                                             |        
|          |`reference`            |The reference array for control normalization                                |
+----------+-----------------------+-----------------------------------------------------------------------------+
|`quantile`|`fixOutliers`\         | Low outlier Meth and Unmeth signals will be fixed\                          |             
|          |`removeBadSamples`\    | Remove bad samples\                                                         |
|          |`badSampleCutoff`\     | The cutoff to label samples as 'bad'\                                       |
|          |`quantileNormalize`\   | Performs quantile normalization\                                            |
|          | `stratified`\         | Performs quantile normalization within region strata\                       |
|          |`mergeManifest`\       | Merged to the output the information in the associated manifest package\    |
|          | \                     |                                                                             |
|          |`sex`                  | Sex of the samples                                                          |
+----------+-----------------------+-----------------------------------------------------------------------------+
|`noob`    | `offset`\             | Offset for the normexp background correct\                                  |
|          | `dyeCorr`\            | Performs dye normalization\                                                 |  
|          | `dyeMethod`           | Dye bias correction to be done                                              |
+---------------+------------------+-----------------------------------------------------------------------------+
|`funnorm` | `nPCs `\              | The number of principal components from the control probes\                 |       
|          | `sex`\                | Sex of the samples\                                                         |      
|          | `bgCorr`\             | Performs NOOB background correction before   functional normalization\      |
|          | \                     |                                                                             |
|          | `dyeCorr`\            | Performs dye normalization\                                                 |
|          | `keepCN`              | Keeps copy number estimates                                                 |
+---------------+-----------------------------+------------------------------------------------------------------+


We can obtain the default settings for each method by invoking  the function `norm_parameters` with no arguments:

```{r}
norm_parameters()
```

However, we can modify the parameter(s) as the following example for `illumina` approach:

```{r}
parameters <- norm_parameters(illumina = list("bg.correct" = FALSE))
parameters$illumina$bg.correct
```

We are going to preprocess the IDAT files and reference panel (\ref{datasets}). We need to specify the IDAT files directory and the reference panel in `RGChannelSet` format. As a result, we will obtain a `GenomicRatioSet` object containing the control and case samples: 

```{r}
GRset <- epi_preprocess(baseDir, reference_panel, pattern = "SampleSheet.csv")
GRset
```


# Epimutations

## Epimutations detection

The function `epimutations` includes 6 methods for epimutation identification: (1) Multivariate Analysis of variance (`manova`), (2) Multivariate Linear Model (`mlm`), (3) isolation forest (`isoforest`), (4) robust mahalanobis distance  (`mahdistmcd`) (5) `quantile` and (6) `beta`. 

To illustrate the following examples we are going to use the dataset `methy` (section \ref{methy}). We need to separate the case and control samples: 

```{r}
case_samples <- methy[,methy$status == "case"]
control_samples <- methy[,methy$status == "control"]
```

We can specify the chromosome or region to analyze which helps to reduce the execution time: 

```{r,  epi_mvo, message=FALSE, warning=FALSE}
epi_mvo <- epimutations(case_samples, control_samples, method = "manova")
```

```{r, epi_method_examples, eval = FALSE}
epi_ml <- epimutations(case_samples, control_samples, method = "mlm")
epi_iso <- epimutations(case_samples, control_samples, method = "isoforest", 
                        chr = "chr10", start = 100, end  = 10000)
epi_mcd <- epimutations(case_samples, control_samples, method = "mahdistmcd", chr = "chr12")
epi_qtl <- epimutations(case_samples, control_samples, method = "quantile", chr = "chr22")
epi_beta <- epimutations(case_samples, control_samples, method = "beta", chr = "chr17")
```


The function `epimutations_one_leave_out()` compared individual methylation profiles of a single
sample (regardless if they are cases or controls) against all other samples from the same cohort. To use this function we do not need to split the dataset: 

```{r eval = FALSE}
#manova (default method)
epi_mva_one_leave_out<- epimutations_one_leave_out(methy)
```

## Unique parameters

The `epi_parameters` function  is  useful to set the individual parameters for each approach. The arguments are described in table \ref{tab:epi_unique_parameter}: 

Table: epimutation function approaches unique parameters
\label{tab:epi_unique_parameter}

+---------------+-----------------------------+----------------------------------------------------------------------+
| Method        | Parameter                   | Description                                                          |
+===============+=============================+======================================================================+
| `manova`\     | `pvalue_cutoff`             | The threshold p-value to select which CpG regions are outliers       |
|  `mlm`\       |                             |                                                                      |
|  `beta`       |                             |                                                                      |
+---------------+-----------------------------+----------------------------------------------------------------------+
| `iso.forest`  | `outlier_score_cutoff`\     | The threshold to select which CpG regions are outliers\              |
|               | `ntrees`                    | The number of binary trees to build for the model                    |
+---------------+-----------------------------+----------------------------------------------------------------------+
|`mahdist.mcd`  | `nsamp`                     | The number of subsets used for initial estimates in the MCD          |
+---------------+-----------------------------+----------------------------------------------------------------------+
|  `quantile`   |`window_sz`\                 |The maximum distance between CpGs to be considered in the same DMR\   |                
|               | \                           |                                                                      |
|               |`offset_mean`/`offset_abs`   | The upper and lower threshold to consider a CpG an outlier           |
+---------------+-----------------------------+----------------------------------------------------------------------+
| `beta`        | `pvalue_cutoff`\            | The minimum p-value to consider a CpG an outlier\                    |
|               | `diff_threshold`            | The minimum methylation difference between the CpG and the\          |
|               |                             |  mean methylation to consider a position an outlier                  |
+---------------+-----------------------------+----------------------------------------------------------------------+



`epi_parameters` with no arguments, returns a list of the default settings for each method: 

```{r}
epi_parameters()
```

The set up of any parameter can be done as the following example for  `manova`: 

```{r}
parameters <- epi_parameters(manova = list("pvalue_cutoff" = 0.01))
parameters$manova$pvalue_cutoff
```


## Results description

The `epimutations` function returns a data frame (tibble) containing all the epimutations identified in the given case sample. If no epimutation is found, the function returns a row containing the case sample information and missing values for all other arguments. The table \ref{tab:output_description} describes each argument: 


Table: epimutation function output arguments description
\label{tab:output_description}

+-----------------------+-----------------------------------------------------------------------------------------------------------------------+
| Column name           | Description                                                                                                           |
+=======================+=======================================================================================================================+
| `epi_id`              | Systematic name for each epimutation identified                                                                       |
+-----------------------+-----------------------------------------------------------------------------------------------------------------------+
| `sample`              | The name of the sample containing that epimutation                                                                    |
+-----------------------+-----------------------------------------------------------------------------------------------------------------------+
| `chromosome`          | The location of the epimutation                                                                                       |
|`start`                |                                                                                                                       |
|`end`                  |                                                                                                                       |
+-----------------------+-----------------------------------------------------------------------------------------------------------------------+
| `sz`                  | The window's size of the event                                                                                        |
+-----------------------+-----------------------------------------------------------------------------------------------------------------------+
| `cpg_n`               | The number of CpGs in the epimutation                                                                                 |
+-----------------------+-----------------------------------------------------------------------------------------------------------------------+
| `cpg_ids`             | The names of CpGs in the epimutation                                                                                  |
+-----------------------+-----------------------------------------------------------------------------------------------------------------------+
| `outlier_score`       | For method `manova` it provides the approximation to F-test and the Pillai score, separated by `/`\                   |
|                       | For method `mlm` it provides the approximation to F-test and the R2 of the model, separated by `/`\                   |
|                       | For method `isoforest` it provides the magnitude of the outlier score.\                                               |
|                       | For method `beta` it provides the mean p-value of all GpGs in that DMR\                                               |
|                       | For methods `quantile` and `mahdistmcd` it is filled with `NA`.                                                       |
+-----------------------+-----------------------------------------------------------------------------------------------------------------------+
| `pvalue`              | For methods `manova` and  `mlm` it provides the p-value obtained from the model.\                                     |
|                       | For method `quantile`, `isoforest`, `beta` and `mahdistmcd` it is filled with `NA`.                                   |
+-----------------------+-----------------------------------------------------------------------------------------------------------------------+ 
|`outlier_direction`    |    Indicates the direction of the outlier with "hypomethylation" and "hypermethylation".\                             | 
|                       | For `manova`, `mlm`, `isoforest`, and `mahdistmcd` \                                                                  |
|                       |    it is computed from the values obtained from `bumphunter`.\                                                        |
|                       | For `beta` is computed from the p value for each \                                                                    |
|                       | CpG using `diff_threshold` and `pvalue_threshold` arguments.\                                                         |
|                       | For `quantile` it is computed from the location of the sample \                                                       |
|                       |    in the reference distribution (left vs. right outlier).                                                            |
+-----------------------+-----------------------------------------------------------------------------------------------------------------------+
| `adj_pvalue`          | For methods `manova` and  `mlm` it provides the adjusted p-value with  \                                              |
|                       | Benjamini-Hochberg based on the total number of regions detected by Bumphunter.     \                                 |
|                       | For method `quantile`, `isoforest`, `mahdistmcd` and `beta` it is filled with `NA`.                                   |
+-----------------------+-----------------------------------------------------------------------------------------------------------------------+ 
| `epi_region_id`       | Name of the epimutation region as defined in `candRegsGR`.                                                            |
+-----------------------+-----------------------------------------------------------------------------------------------------------------------+
| `CRE`                 | cREs (cis-Regulatory Elements) as defined by ENCODE overlapping the epimutation region.                               |
+-----------------------+-----------------------------------------------------------------------------------------------------------------------+
| `CRE_type`            | Type of cREs (cis-Regulatory Elements) as defined by ENCODE.                                                          |
+-----------------------+-----------------------------------------------------------------------------------------------------------------------+

As an example we are going to visualize the obtained results with MANOVA method  (`epi_mvo`): 

```{r}
dim(epi_mvo)
class(epi_mvo)
colnames(epi_mvo)
head(as.data.frame(epi_mvo[,1:6]), 1)
```

## Epimutations annotations

The `annotate_epimutations` function  enriches the identified epimutations. It includes information about GENECODE gene names, description of the regulatory feature provided by methylation consortium, the location of the CpG relative to the CpG island, OMIM accession and description number and Ensembl region id, coordinates, type and tissue:

```{r, ann, message=FALSE, tidy = TRUE}
rst_mvo <- annotate_epimutations(epi_mvo)
```

```{r}
colnames(rst_mvo[2:3, c(1, 12:14)])
```

```{r ann results, eval = FALSE}
rst_mvo[c(1,29), c(1, 12:14)]```

```{r echo = FALSE}
library(knitr)
kable(rst_mvo[c(1,29), c(1, 12:14)], caption = "epimutations annotation")
```


+---------------------------+-----------------------------------------------------------------------------------------+
| Column name               | Description                                                                             |
+===========================+=========================================================================================+
| `GencodeBasicV12_NAME`    | Gene names from the basic GENECODE build                                                |
+---------------------------+-----------------------------------------------------------------------------------------+
| `Regulatory_Feature_Group`| Description of the regulatory feature provided by the Methylation Consortium            |
+---------------------------+-----------------------------------------------------------------------------------------+
| `Relation_to_Island`      | The location of the CpG relative to the CpG island                                      |
+---------------------------+-----------------------------------------------------------------------------------------+
| `OMIM_ACC` \              | OMIM accession and description number                                                   |
| `OMIM_DESC`               |                                                                                         |
+---------------------------+-----------------------------------------------------------------------------------------+
| `ensembl_reg_id` \        | The Ensembl region id, coordinates, type and tissue                                     |
|`ensembl_reg_coordinates`\ |                                                                                         |
|`ensembl_reg_type`\        |                                                                                         |
|`ensembl_reg_tissues`      |                                                                                         |
+---------------------------+-----------------------------------------------------------------------------------------+

## Epimutation visualization

The `plot_epimutations` function locates the epimutations along the genome.  It plots the methylation values of the case sample in red, the control samples in dashed black lines and population mean in blue(figure \ref{fig:plot_default}). 

```{r, plot_default, eval = FALSE}
knitr::opts_chunk$set(fig.pos = 'H')
plot_epimutations(as.data.frame(epi_mvo[1,]), methy)
```

```{r, fig_plot_default, echo = FALSE, fig.cap = "beta values of case sample againts control samples \\label{fig:plot_default}", fig.align = 'center', fig.pos = "H"}
knitr::include_graphics(file.path(getwd(),"fig/plot_example_beta_values.PNG"))
``` 

If we set the argument `gene_annot == TRUE` the plot includes the gene annotations: 
 (figure \ref{fig:plot_mvo_genes_annot}): 

```{r, plot_genes_annot, eval = FALSE}
knitr::opts_chunk$set(fig.pos = 'H')
plot_epimutations(as.data.frame(epi_mvo[1,]), methy, genes_annot = TRUE)
```

```{r, fig_plot_genes_annot, echo = FALSE, fig.cap = "beta values of case sample againts control samples including UCSC gene annotation \\label{fig:plot_mvo_genes_annot}", fig.align = 'center', fig.pos = "H"}
knitr::include_graphics(file.path(getwd(),"fig/plot_example_beta_values_UCSC.PNG"))
``` 

To  plot the chromatin marks H3K4me3, H3K27me3  and H3K27ac we need to specify the argument  `regulation = TRUE`(figure \ref{fig:plot_mvo_regulation}): 

* **H3K4me3**: commonly associated with the activation of transcription of nearby genes.
* **H3K27me3**: is used in epigenetics to look for inactive genes.
* **H3K27ac**: is associated with the higher activation of transcription and therefore defined as an active enhancer mark


```{r, plot_mvo_regulation, eval = FALSE}
knitr::opts_chunk$set(fig.pos = 'H')
plot_epimutations(as.data.frame(epi_mvo[1,]), methy, regulation = TRUE)
```

```{r, fig_plot_genes_annot, echo = FALSE, fig.cap = "beta values of case sample againts control samples including CpG islands  \\label{fig:plot_mvo_regulation}", fig.align = 'center', fig.pos = "H"}
knitr::include_graphics(file.path(getwd(),"fig/plot_example_beta_Values_ucsc_cpg_island.PNG"))
``` 


# Results

## Simulations

We explored the effect of sample size on the identification of 4 epimutations experimentally validated by Grag and colleagues [@garg2020survey].   We downloaded the data from [GEO](https://www.ncbi.nlm.nih.gov/geo/).  We accessed DNA methylation data from a total of 1, 417 individuals from [GSE51032](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE51032) and 
[GSE111629](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi) cohorts. 

We evaluated the performance of the methods using True Positive Rate (TPR), False Positive Rate (FPR) and accuracy. We used the TPR to measure the proportion of detected epimutations by our method present in the validated epimutations (table \ref{tab:table_validated}). The FPR computes the identified epimutations by our method that were not found by Grag and colleagues.  The accuracy measures the closeness of the epimutations detected by our methods to the validated epimutations. 


We selected control samples randomly using different sample size: 20, 30, 40, 50, 60, 70, 80, 90 and 100. We set as case samples the individuals containing the validated epimutations (table \ref{tab:table_validated}):  GSM1235784 sample from GSE51032 cohort; and GSM3035933, GSM3035791, GSM3035807 and GSM3035685 samples from GSE111629. We utilized those case samples to compute TPR and accuracy. However, we measured the FPR selecting all the case samples with no epimutations identified by Grag and colleagues. 

We analyzed regions of $\approx$ 20 kb containing $\ge$ 3 GpGs. We executed 100 times the same process for each control sample size.  

```{r tableEpiValidated, echo=FALSE, tab.cap='(ref:tableEpiValidated)'}
load("C:/Users/nla94/Documents/GitHub/Supplementary-Material/Abarrategui_2021/hpc/inputs/epi_validated.rda")
epi_validated <- as.data.frame(epi_validated)
epi_validated <- epi_validated[c(1,1,2,3,4,4),]
epi_validated$delta_beta <- c(0.253, 0.237, 0.185, 0.604, 0.0703, 0.231)
epi_validated$samples <- c("GSM1235784", "GSM3035791",
                           "GSM3035685", "GSM3035933",
                           "GSM3035791", "GSM3035807")
epi_validated <- epi_validated[,c(7,1:6)]
colnames(epi_validated) <- c("Samples", "Chromosome", "Start", "End", 	"Width", "Strand", "Delta-beta")
knitr::kable(epi_validated, format = "markdown", caption = "validated epimutations (Garg et al. 2020).\\label{tab:table_validated}")
```

Figures \ref{fig:graph_GSM1235784}, \ref{fig:graph_GSM3035685}, \ref{fig:graph_GSM3035791}, \ref{fig:graph_GSM3035791.2}, \ref{fig:graph_GSM3035807} and \ref{fig:graph_GSM3035933}  illustrate the  methylation values for each validated epimutation. The y-axis represents the methylation value and the x-axis the location of the epimutation in the genome.  We represented the case sample in red, the control samples in dashed black lines and population mean in blue: 

```{r graph_GSM1235784, echo = FALSE, fig.cap = "GSE51032 cohort samples in the region chr17:46018654-46019184", fig.align = 'center', fig.pos = "H"}
knitr::include_graphics(file.path(getwd(),"fig/GSM1235784_chr17-46018654-46019184.png"))
``` 

```{r graph_GSM3035685, echo = FALSE, fig.cap = "GSE111629 cohort samples in the region chr19:11199851-11200146", fig.align = 'center', fig.pos = "H"}
knitr::include_graphics(file.path(getwd(),"fig/GSM3035685_chr19-11199851-11200146.png"))
``` 

```{r graph_GSM3035791, echo = FALSE, fig.cap = "GSE111629 cohort samples in the region chr5:67584194-67584380", fig.align = 'center', fig.pos = "H"}
knitr::include_graphics(file.path(getwd(),"fig/GSM3035791_chr5-67584194-67584380.png"))
``` 

```{r graph_GSM3035791.2, echo = FALSE, fig.cap = "GSE111629 cohort samples in the region chr17:46018654-46019184", fig.align = 'center', fig.pos = "H"}
knitr::include_graphics(file.path(getwd(),"fig/GSM3035791_chr17-46018654-46019184.png"))
``` 

```{r graph_GSM3035807, echo = FALSE, fig.cap = "GSE111629 cohort samples in the region chr5:67583972-67584380", fig.align = 'center', fig.pos = "H"}
knitr::include_graphics(file.path(getwd(),"fig/GSM3035807_chr5-67583972-67584380.png"))
``` 

```{r graph_GSM3035933, echo = FALSE, fig.cap = "GSE111629 cohort samples in the region chr5:10249761-10251252", fig.align = 'center', fig.pos = "H"}
knitr::include_graphics(file.path(getwd(),"fig/GSM3035933_chr5-10249761-10251252.png"))
``` 



```{r table_GSE51032, echo = FALSE}
library(dplyr)
load("C:/Users/nla94/Documents/GitHub/Supplementary-Material/Abarrategui_2021/result_files/3-TPR_FPR_accuracy/TPR_FPR_accuracy_GSE51032.rda")
df <- df$chr17_46018653_46019185
df[,3:5] <- apply(df[,c(-1, -2)], 2,as.numeric)
#kableExtra::kbl(list(df[1:30,], df[31:54,]), caption = "GSE51032 cohort TPR, FPR and accuracy")
kableExtra::kbl(list(df[1:30,], df[31:54,])) %>%
  kableExtra::kable_paper("striped", full_width = F) %>%
  kableExtra::kable_styling(latex_options = "HOLD_position")
#kableExtra::kbl(df[,-2], align = 'c') %>%
#    kableExtra::kable_paper("striped", full_width = F) %>%
#    kableExtra::pack_rows(index = c("n20" = 6, "n30" = 6, "n40" = 6, 
#                          "n50" = 6, "n60" = 6, "n70" = 6, 
#                          "n80" = 6, "n90" = 6, "n100" = 6),
#              label_row_css = "background-color: #666; color: #fff;")
#knitr::kable(mean_table, align = "lccrr", caption = "`epimutations` function TPR, FPR and accuracy using GSE51032 cohort")
```

```{r table_GSE111629, echo = FALSE, results='asis'}
load("C:/Users/nla94/Documents/GitHub/Supplementary-Material/Abarrategui_2021/result_files/3-TPR_FPR_accuracy/TPR_FPR_accuracy_GSE111629.rda")
df <- as.data.frame(data.table::rbindlist(df))
df[,3:5] <- apply(df[,c(-1, -2)], 2,as.numeric)
mean_table <- aggregate( .~ method + n, data = df, FUN= "mean")
#kableExtra::kbl(list(mean_table[1:30,], mean_table[31:54,]), 
                #caption = "GSE111629 cohort TPR, FPR and accuracy. The measures shown are the mean for all the #validated epimutations identified.")
kableExtra::kbl(list(mean_table[1:30,], mean_table[31:54,])) %>%
  kableExtra::kable_paper("striped", full_width = F) %>%
  kableExtra::kable_styling(latex_options = "HOLD_position")
#kableExtra::kbl(mean_table, align = 'c') %>%
#    kableExtra::kable_paper("striped", full_width = F) %>%
#    kableExtra::pack_rows(index = c("n20" = 6, "n30" = 6, "n40" = 6, 
#                          "n50" = 6, "n60" = 6, "n70" = 6, 
#                          "n80" = 6, "n90" = 6, "n100" = 6),
#              label_row_css = "background-color: #666; color: #fff;")
#knitr::kable(mean_table, align = "lccrr", caption = "`epimutations` function TPR, FPR and accuracy for GSE111629 cohort. The measures shown are the mean for all the validated epimutations identified.")
```

## Methods testing

We run the six implemented methods in `epimutacions`  and the  Perl script (quantile-perl) from  Garg and colleagues. We implemented that script in R (Quantile-R) with a small difference.  Quantile-R excludes the target sample when computing the methylation quantiles while in quantile-perl the quantiles are computed including the target sample. We used GSE84727 dataset from GEO which  contains a total of  847 whole blood adults samples, 414 schizophrenia cases and 433 controls. This dataset have been previously analyzed by Garg and colleagues. 

The methods based in bumphunter detected epimutations in almost all samples.  However, beta identified epimutations in $\approx$ 75$\%$ of the samples and quantile in $\approx$ 33$\%$  (figure \ref{fig:overlap_methods}). All the epimutations detected by Garg and colleagues are found at least by one of our methods. Isolation forest, mlm and manova shared >80$\%$ of the epimutations detected. They also identified  most of the epimutations found by beta, being only  10$\%$ of the epimutations specific to beta. Nevertheless, mahalanobis distance presented more divergent results, the $\approx$ 38$\%$ of the detected epimutations  are not present in the other methods. 

Manova, mlm and isolation forest detected several epimutations per individual. Quantile found $\leq$ 1 epimutation per individual. However,  beta and mahalanobis methods identified few epimutations (figure \ref{fig:prop_individuals}).  

```{r overlap_methods, echo = FALSE, fig.cap = "Overlap between methods", fig.align='center'}
knitr::include_graphics(file.path(getwd(),"fig/methods.png"))
```

 
```{r prop_individuals, echo = FALSE, fig.cap = "Proportion of individuals with epimutations", fig.align='center'}
knitr::include_graphics(file.path(getwd(),"fig/individuals.png"))
```


\newpage

# Acknowledgements

We acknowledge the organizers of the [European BioHackathon 2020](https://www.biohackathon-europe.org/) for their support.

All the team members of *Project #5* for the contribution to this package: 

| Name | Surname | ORCID | Affiliation | Team |
|:-----|:--------|:-----:|:------------|:-----|
| Leire | Abarrategui | 0000-0002-1175-038X | Faculty of Medical Sciences, Newcastle University, Newcastle-Upon-Tyne, UK; Autonomous University of Barcelona (UAB), Barcelona, Spain | Development |
| Lordstrong | Akano | 0000-0002-1404-0295 | College of Medicine, University of Ibadan | Development |
| James | Baye | 0000-0002-0078-3688 | Wellcome/MRC Cambridge Stem Cell Institute, University of Cambridge, Cambridge CB2 0AW, UK; Department of Physics, University of Cambridge, Cambridge CB2 3DY, UK | Development |
| Alejandro | Caceres | - | ISGlobal, Barcelona Institute for Global Health, Dr Aiguader 88, 08003 Barcelona, Spain; Centro de InvestigaciÃ³n BiomÃ©dica en Red en EpidemiologÃ­a y Salud PÃºblica (CIBERESP), Madrid, Spain | Development |
| Carles | Hernandez-Ferrer | 0000-0002-8029-7160 | Centro Nacional de AnÃ¡lisis GenÃ³mico (CNAG-CRG), Center for Genomic, Regulation; Barcelona Institute of Science and Technology (BIST), Barcelona, Catalonia, Spain | Development |		
| Pavlo | Hrab | 0000-0002-0742-8478 | Department of Genetics and Biotechnology, Biology faculty, Ivan Franko National University of Lviv | Validation |
| Raquel | Manzano | 0000-0002-5124-8992 | Cancer Research UK Cambridge Institute; University of Cambridge, Cambridge, United Kingdom | Reporting |
| Margherita | Mutarelli | 0000-0002-2168-5059 | Institute of Applied Sciences and Intelligent Systems (ISASI-CNR) | Validation |
| Carlos | Ruiz-Arenas | 0000-0002-6014-3498 | Centro de InvestigaciÃ³n BiomÃ©dica en Red de Enfermedades Raras (CIBERER), Barcelona, Spain; Universitat Pompeu Fabra (UPF), Barcelona, Spain | Reporting |

# Session Info

```{r}
sessionInfo()
```

# References
