---
title: "epimutacions: a Bioconductor package to identify outliers in rare diseases DNA methylation data"
author:
  - Leire Abarrategui 
  - Carlos Ruiz-Arenas
  - Carles Hernandez-Ferrer
  - Juan R. Gonzalez
  - Patricia Ryser-Welch
date: "`r Sys.Date()`"
header-includes:
   - \renewcommand{\and}{\\}
   - \usepackage{floatrow}
   - \floatsetup[figure]{capposition=top}
output:
  pdf_document: 
    latex_engine: xelatex
    fig_caption: true
    number_sections: true

    
subtitle: "Supplementary material" 
vignette: >
  %\VignetteIndexEntry{Detection of epimutations with state of the art methods in methylation data} 
  %\VignetteEngine{knitr::rmarkdown} 
  %\VignetteEncoding{UTF-8}
bibliography: refs.bib
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(comment="", warning = FALSE, message = FALSE, cache = TRUE)
```
\newpage
\tableofcontents
\newpage

# Introduction

## Background

According to the European Commission, rare diseases are pathologies with a prevalence of less than 1 person in 2,000 people [@EU_RD]. Most of these conditions have an onset during childhood and a strong genetic etiology [@lopez2016social]. Consequently, rare disease diagnosis has relied on identifying genetic and genomic mutations that can cause the disease [@aref2019]. Although these variants have provided a diagnosis for many patients and families, around 60% of the cases remained undiagnosed [@lionel2018improved]. Aberrant methylation can be an underlying cause of undiagnosed patients, either as a primary event (a.k.a. epimutation) or as a functional consequence of chromatin dysregulation by genetic or environmental agents (a.k.a. episignature)  [@aref2019]. Epimutations are the cause of some rare diseases, such as Prader-Willi, Angelman or Beckwith-Wiedemann syndromes [@aref2019] and some human malformations [@serra2015dna]. Syndrome-specific episignatures are increasingly defined as biomarkers for a growing number of disorders [@aref2019; @garg2020survey]. Therefore, tools to detect epimutations and episignatures should be made available to the rare disease community and included in standardized analysis workflows.


This manual describes the tools available in  `epimutacions` package to identify epivariants using multiple outlier detection  approaches. It also includes functions to plot and annotate the epimutations. 

The name of the package is `epimutacions` (pronounced `ɛ pi mu ta 'sj ons`) which means epimutations in Catalan, a language from the northeast of Spain.

## Methodology

The `epimutacions` package computes a genome-wide DNA methylation analysis to detect the epigenetic variants to be considered as biomarkers for samples with rare diseases (epimutations). It compares a case sample with suspected rare disease against a reference panel.  The package focused on the detection of outliers in DNA methylation patterns associated with the diseases as proposed by [@aref2019].  

The identification of relevant genomic methylation regions for a given sample having a rare disease will be driven by detecting differentially methylated CpG sites when comparing beta values of all control samples with the given proband. Firstly, bump-hunter [@jaffe2012bump] approach is used to identify the Differentially Methylated Regions (DMRs). Then, CpGs in the proband sample are tested in those DMRs in order to identify regions with CpGs being outliers when comparing with the reference panel. To this end, different anomaly detection statistical approaches are used. These include Multivariate Analysis of Variance (MANOVA) [@friedrich2017package], Multivariate Linear Model [@martin2020multivariate], isolation forest [@cortes2021package] and robust mahalanobis distance [@maechler2021package]. However, quantile [@garg2020survey] and Beta methods do not use bump-hunter output. Quantile [@garg2020survey] checks for each CpG, if the proband’s measurement is an outlier. Then, it calls an epimutation to those regions where 3 contiguous CpGs are outliers, and they are separated by less than 500 base pairs. Beta approach models the DNA methylation data using a beta distribution.  

## Input data

The package allows two different types of inputs: 

 * (1) Case samples `IDAT` files (raw microarray intensities) together with `RGChannelSet` class object as reference panel. The reference panel can be supplied by the user or can be selected through the example datasets available in the package. 
 
* (2) `GenomicRatioSet` class object containing case and control samples. 

The input data should contain information about $\beta$ values of CpG sites, phenotype and feature data. 

The normalization converts the raw microarray intesities into usable methylation measurement ($\beta$ values at CpG locus level). As a results, we obtain  `GenomicRatioSet` object that can be used as `epimutations()` function input.  The case and control samples need to be introduced separately in `epimutations()` function.  

```{r workflow, echo=FALSE, fig.cap="Allowed data formats, normalization and epimutations() function input", out.width = '90%', fig.align='center'}
knitr::include_graphics(file.path(getwd(),"vignette/fig/workflow.png"))
```



# Getting started

The `epimutacions` package is available on GitHub and can be installed by executing the following command:

```{r, eval=FALSE}
install_github("isglobal-brge/epimutacions")
```

The package is loaded in R as usual:

```{r, message=FALSE}
library(epimutacions)
```

The document has the following dependencies: 

```{r, eval=FALSE}
library(Knitr)
library(kableExtra)
```

\newpage

# Datasets


## Candidate regions

Epimutations detection has two main steps: (1) definition of candidate regions and (2) evaluation of outlier significance. Although there are different algorithms to define epimutations regions, they share common features. In general, we define an epimutation as at least 3 contiguous CpGs with a maximum distance of 1kb between them [@aref2019; @garg2020survey].  

In Illumina 450K array [@reproducibilityinfinium], probes are unequally distributed along the genome, limiting the number of regions that can fulfil the requirements to be considered an epimutation. So, we have computed a dataset containing the regions that are candidates to become an epimutation.  

To define the candidate epimutations, we relied on the clustering from bumphunter [@jaffe2012bump]. We defined a primary dataset with all the CpGs from the Illumina 450K array.  Then, we run bumphunter and selected those regions with at least 3 CpGs. As a result, we found 40408 candidate epimutations which are available in `candRegsGR` dataset.

Besides, we converted the candidate region from hg19 to hg38 coordinates, using NCBI remap [@holmes2020spdi]. We selected regions that mapped to one region in hg38 with the same length. This yielded a total of 39944, the 98.85% of total hg19 regions. After converting to hg38, we can use these ranges to be annotated to ENCODE cREs [@hon2020expanded]. Overall, we mapped 30163 candidate regions to cREs, representing 74.65% of total candidate regions. 


```{r, message = FALSE}
#Candidate regions dataset
##load data
data("candRegsGR")
##data class
class(candRegsGR)
##dataset
candRegsGR
```

## GenomicRatioSet

The package includes a small `GenomicRatioSet` class dataset (`methy` `r data("methy", package = "epimutacions")`) containing the DNA methylation profiles from a total of  `r ncol(methy)` individuals, `r ncol(methy[,methy$status == "case"])` cases and `r ncol(methy[,methy$status == "control"])` controls. The DNA methylation profiles were generated using the Illumina 450k Human Methylation BeadChip [@reproducibilityinfinium]. The data corresponds to GSE104812 cohort from  [Gene Expression Omnibus (GEO)](https://www.ncbi.nlm.nih.gov/geo/). It has been adapted for package usage. 


```{r}
data("methy")
methy

table(methy$status)
```

## IDAT files and RGChannelSet

Additionally, as external data  4  case samples IDAT files from [GSE131350](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE131350) cohort are included.

```{r IDAT files}
baseDir <- system.file("extdata", package = "epimutacions")
list.files(baseDir)
targets <- minfi::read.metharray.sheet(baseDir)
```

The reference panel (`RGChannelSet` object) contains DNA methylation profiles of 22 whole cord blood samples from healthy children born via caesarian [(GEO: GSE127824)](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE12782). This dataset is  available in the `ExperimentHub` package. 



# Preprocessing

The normalization removes the unwanted variation caused by the batch effect when combining data from different sources. The preprocessing in `epimutacions` package is done by `epi_preprocess()` function. It contains 6 preprocessing methods  corresponding to minfi package [@aryee2014minfi], which can be selected by the user:
   


+---------------+-----------------------------+----------------------------------------------------------------------+
| Method        |  Function                   | Description                                                          |
+===============+=============================+======================================================================+
| `raw`         | `preprocessRaw`             | Converts the Red/Green channel for an Illumina methylation           |  
|               |                             |  array into methylation signal. This method does not normalize       |
|               |                             |  the data.                                                           |
+---------------+-----------------------------+----------------------------------------------------------------------+
| `illumina  `  | `preprocessIllumina`        |  Implements preprocessing for Illumina methylation                   |
|               |                             |      microarrays as used in Genome Studio.                           |
+---------------+-----------------------------+----------------------------------------------------------------------+
|`swan`         | `preprocessSWAN `           |Subset-quantile Within Array Normalisation (SWAN). It allows Infinium |
|               |                             | I and II type probes on a single array to be normalized together.    |
+---------------+-----------------------------+----------------------------------------------------------------------+
|  `quantile`   |`preprocessQuantile`         |Implements stratified quantile normalization preprocessing for        |                
|               |                             |Illumina methylation microarrays.                                     |
+---------------+-----------------------------+----------------------------------------------------------------------+
| `noob`        | `preprocessNoob`            | Noob (normal-exponential out-of-band) is a background correction     | 
|               |                             | method with dye-bias normalization for                               |
|               |                             | Illumina Infinium methylation arrays.                                |
+---------------+-----------------------------+----------------------------------------------------------------------+
| `funnorm`     | `preprocessFunnorm `        | Functional normalization (FunNorm) is a between-array                |
|               |                             | normalization method for the Illumina Infinium                       |
|               |                             | HumanMethylation450 platform.                                        | 
+---------------+-----------------------------+----------------------------------------------------------------------+

Table: Preprocessing methods description
\label{tab:prep_des}

Those methods have unique parameters (table \ref{tab:prep_unique_parameter}) that can be defined through `norm_parameters()` function: 

Table: Preprocessing methods unique parameters
\label{tab:prep_unique_parameter}

+----------+-----------------------+-----------------------------------------------------------------------------+
| Method   |Parameters             |Description                                                                  |
+==========+=======================+=============================================================================+
|`illumina`|`bg.correct`\          |Performs background correction\                                              |
|          |`normalize`\           |Performs controls normalization\                                             |        
|          |`reference`            |The reference array for control normalization                                |
+----------+-----------------------+-----------------------------------------------------------------------------+
|`quantile`|`fixOutliers`\         | Low outlier Meth and Unmeth signals will be fixed\                          |             
|          |`removeBadSamples`\    | Remove bad samples\                                                         |
|          |`badSampleCutoff`\     | The cutoff to label samples as 'bad'\                                       |
|          |`quantileNormalize`\   | Performs quantile normalization\                                            |
|          | `stratified`\         | Performs quantile normalization within region strata\                       |
|          |`mergeManifest`\       | Merged to the output the information in the associated manifest package\    |
|          | \                     |                                                                             |
|          |`sex`                  | Sex of the samples                                                          |
+----------+-----------------------+-----------------------------------------------------------------------------+
|`noob`    | `offset`\             | Offset for the normexp background correct\                                  |
|          | `dyeCorr`\            | Performs dye normalization\                                                 |  
|          | `dyeMethod`           | Dye bias correction to be done                                              |
+---------------+------------------+-----------------------------------------------------------------------------+
|`funnorm` | `nPCs `\              | The number of principal components from the control probes\                 |       
|          | `sex`\                | Sex of the samples\                                                         |      
|          | `bgCorr`\             | Performs NOOB background correction before   functional normalization\      |
|          | \                     |                                                                             |
|          | `dyeCorr`\            | Performs dye normalization\                                                 |
|          | `keepCN`              | Keeps copy number estimates                                                 |
+---------------+-----------------------------+------------------------------------------------------------------+



The default settings for each method can be obtained by invoking  the function `norm_parameters()` with no arguments:

```{r}
norm_parameters()
```

However, to modify the parameters related to a  method you can do as the following example for `illumina` approach:

```{r}
parameters <- norm_parameters(illumina = list("bg.correct" = FALSE))
parameters$illumina$bg.correct
```

# Epimutations

## Epimutations detection

The `epimutacions` package includes 6 methods for epivariants identification: (1) [Multivariate Analysis of variance (`manova`)](https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/manova), (2) [Multivariate Linear Model (`mlm`)](https://github.com/dgarrimar/mlm/blob/master/R/mlm.R), (3) [isolation forest (`isoforest`)](https://rdrr.io/cran/isotree/man/isolation.forest.html), (4) [robust mahalanobis distance  (`mahdistmcd`)](https://www.rdocumentation.org/packages/robustbase/versions/0.93-8/topics/covMcd) (5) `quantile` and (6) `beta`. 


In the mentioned first 4 methods, firstly, Differentially Methylated Regions (DMRs) are identified using bump-hunter method [@jaffe2012bump]. Then, those DMRs are tested to identify regions with CpGs being outliers when comparing with the reference panel. However, `quantile` and `beta` do not identify outliers by filtering the DMRs. `quantile`  utilized a sliding window approach to individually compare the methylation value  in each proband against the reference panel. `Beta` used beta distribution to identify epivariants in the case sample. 

To ilustrate multiple examples we are going to use `methy` dataset. Fristly, we are going to split the dataset in two subset: (1) cases (`case_samples`) and (2) controls (`control_samples`): 

```{r}
case_samples <- methy[,methy$status == "case"]
control_samples <- methy[,methy$status == "control"]
```

Then, we are going to identify epimutations in each case sample using one statistical approach at a time: 

```{r, epimutations_detection, message=FALSE, warning=FALSE}
epi_mvo <- epimutations(case_samples, control_samples, method = "manova")
epi_ml <- epimutations(case_samples, control_samples, method = "mlm")
epi_iso <- epimutations(case_samples, control_samples, method = "isoforest")
epi_mcd <- epimutations(case_samples, control_samples, method = "mahdistmcd")
epi_qtl <- epimutations(case_samples, control_samples, method = "quantile")
epi_beta <- epimutations(case_samples, control_samples, method = "beta")
```




## Unique parameters

The `epi_parameters()` function  is  useful to set the individual parameters for each approach. The arguments are described in table \ref{tab:epi_unique_parameter}: 

Table: epimutation function approaches unique parameters
\label{tab:epi_unique_parameter}

+---------------+-----------------------------+----------------------------------------------------------------------+
| Method        | Parameter                   | Description                                                          |
+===============+=============================+======================================================================+
| `manova`\     | `pvalue_cutoff`             | The threshold p-value to select which CpG regions are outliers       |
|  `mlm`\       |                             |                                                                      |
|  `beta`       |                             |                                                                      |
+---------------+-----------------------------+----------------------------------------------------------------------+
| `iso.forest`  | `outlier_score_cutoff`\     | The threshold to select which CpG regions are outliers\              |
|               | `ntrees`                    | The number of binary trees to build for the model                    |
+---------------+-----------------------------+----------------------------------------------------------------------+
|`mahdist.mcd`  | `nsamp`                     | The number of subsets used for initial estimates in the MCD          |
+---------------+-----------------------------+----------------------------------------------------------------------+
|  `quantile`   |`window_sz`\                 |The maximum distance between CpGs to be considered in the same DMR\   |                
|               | \                           |                                                                      |
|               |`offset_mean`/`offset_abs`   | The upper and lower threshold to consider a CpG an outlier           |
+---------------+-----------------------------+----------------------------------------------------------------------+
| `beta`        | `pvalue_cutoff`\            | The minimum p-value to consider a CpG an outlier\                    |
|               | `diff_threshold`            | The minimum methylation difference between the CpG and the\          |
|               |                             |  mean methylation to consider a position an outlier                  |
+---------------+-----------------------------+----------------------------------------------------------------------+



Invoking `epi_parameters()` with no arguments returns a list of the default settings for each method: 

```{r}
epi_parameters()
```

The set up of any parameter can be done as the following example of p-value cut-off for  `manova`: 

```{r}
parameters <- epi_parameters(manova = list("pvalue_cutoff" = 0.01))
parameters$manova$pvalue_cutoff
```


## Results description

The `epimutations` function returns a `tibble` containing all the epivariants identified in the given case sample. In case no epimutation is found, a row containing the case sample information and missing values for each argument is returned. Table \ref{tab:output_description} describes each argument in the result data frame: 


Table: epimutation function output arguments description
\label{tab:output_description}

+-----------------------+-----------------------------------------------------------------------------------------------------------------------+
| Column name           | Description                                                                                                           |
+=======================+=======================================================================================================================+
| `epi_id`              | Systematic name for each epimutation identified                                                                       |
+-----------------------+-----------------------------------------------------------------------------------------------------------------------+
| `sample`              | The name of the sample containing that epimutation                                                                    |
+-----------------------+-----------------------------------------------------------------------------------------------------------------------+
| `chromosome`          | The location of the epimutation                                                                                       |
|`start`                |                                                                                                                       |
|`end`                  |                                                                                                                       |
+-----------------------+-----------------------------------------------------------------------------------------------------------------------+
| `sz`                  | The window's size of the event                                                                                        |
+-----------------------+-----------------------------------------------------------------------------------------------------------------------+
| `cpg_n`               | The number of CpGs in the epimutation                                                                                 |
+-----------------------+-----------------------------------------------------------------------------------------------------------------------+
| `cpg_ids`             | The names of CpGs in the epimutation                                                                                  |
+-----------------------+-----------------------------------------------------------------------------------------------------------------------+
| `outlier_score`       | For method `manova` it provides the approximation to F-test and the Pillai score, separated by `/`\                   |
|                       | For method `mlm` it provides the approximation to F-test and the R2 of the model, separated by `/`\                   |
|                       | For method `isoforest` it provides the magnitude of the outlier score.\                                               |
|                       | For method `beta` it provides the mean p-value of all GpGs in that DMR\                                               |
|                       | For methods `quantile` and `mahdistmcd` it is filled with `NA`.                                                       |
+-----------------------+-----------------------------------------------------------------------------------------------------------------------+
| `pvalue`              | For methods `manova` and  `mlm` it provides the p-value obtained from the model.\                                     |
|                       | For method `quantile`, `isoforest`, `beta` and `mahdistmcd` it is filled with `NA`.                                   |
+-----------------------+-----------------------------------------------------------------------------------------------------------------------+ 
|`outlier_direction`    |    Indicates the direction of the outlier with "hypomethylation" and "hypermethylation".\                             | 
|                       | For `manova`, `mlm`, `isoforest`, and `mahdistmcd` \                                                                  |
|                       |    it is computed from the values obtained from `bumphunter`.\                                                        |
|                       | For `beta` is computed from the p value for each \                                                                    |
|                       | CpG using `diff_threshold` and `pvalue_threshold` arguments.\                                                         |
|                       | For `quantile` it is computed from the location of the sample \                                                       |
|                       |    in the reference distribution (left vs. right outlier).                                                            |
+-----------------------+-----------------------------------------------------------------------------------------------------------------------+
| `adj_pvalue`          | For methods `manova` and  `mlm` it provides the adjusted p-value with  \                                              |
|                       | Benjamini-Hochberg based on the total number of regions detected by Bumphunter.     \                                 |
|                       | For method `quantile`, `isoforest`, `mahdistmcd` and `beta` it is filled with `NA`.                                   |
+-----------------------+-----------------------------------------------------------------------------------------------------------------------+ 
| `epi_region_id`       | Name of the epimutation region as defined in `candRegsGR`.                                                            |
+-----------------------+-----------------------------------------------------------------------------------------------------------------------+
| `CRE`                 | cREs (cis-Regulatory Elements) as defined by ENCODE overlapping the epimutation region.                               |
+-----------------------+-----------------------------------------------------------------------------------------------------------------------+
| `CRE_type`            | Type of cREs (cis-Regulatory Elements) as defined by ENCODE.                                                          |
+-----------------------+-----------------------------------------------------------------------------------------------------------------------+



## Epimutations annotations

The `epimutacions` package also includes the `annotate_epimutations()` function dedicated to enriching the epimutations identified by the previously described methods:

```{r, ann, message=FALSE, tidy = TRUE}
rst_mvo <- annotate_epimutations(epi_mvo)
```

```{r ann results, eval = FALSE}
rst_mvo[1:2, c(1, 12:14)]
```

```{r echo = FALSE}
library(knitr)
kable(rst_mvo[c(1,29), c(1, 12:14)], caption = "epimutations annotation")
```


## Epimutation visualization

The  visualization approach locates the epimutations along the genome. The function `plot_epimutations()` plots the methylation values of the  individual with the epimutation in red, the control samples in dashed black lines and population mean in blue (figure \ref{fig:plot_default}):  

```{r plot_default, fig.cap="beta values of case sample againts control samples \\label{fig:plot_default}"}
plot_epimutations(as.data.frame(epi_mvo[1,]), methy)
```

Furthermore, it includes the gene annotations in the regions in which the epivariation is located.  This can be achieved by using the argument  `gene_annot == TRUE` (figure \ref{fig:plot_mvo_genes_annot}): 

```{r, plot_mvo_genes_annot, fig.cap="beta values of case sample againts control samples including UCSC gene annotation \\label{fig:plot_mvo_genes_annot}"}
plot_epimutations(as.data.frame(epi_mvo[1,]), methy, genes_annot = TRUE)
```


Also, it is possible to plot (figure \ref{fig:plot_mvo_regulation}) CpG islands and chromatin marks H3K4me3, H3K27me3  and H3K27ac by setting the argument `regulation = TRUE`:

* **H3K4me3**: commonly associated with the activation of transcription of nearby genes.
* **H3K27me3**: is used in epigenetics to look for inactive genes.
* **H3K27ac**: is associated with the higher activation of transcription and therefore defined as an active enhancer mark


```{r, plot_mvo_regulation, eval = FALSE}
plot_epimutations(as.data.frame(epi_mvo[1,]), methy, regulation = TRUE)
```

```{r plot_mvo_regulation_fig, echo=FALSE, fig.cap = "beta values of case sample againts control samples including CpG islands  \\label{fig:plot_mvo_regulation}", out.width = '90%', fig.align='center'}
knitr::include_graphics(file.path(getwd(),"vignette/fig/plot_regulation.JPG"))
```

# Method validation

We validate our method (`epimutacions`) reproducing the results in the study [@garg2020survey]. 

## Data collection 

The data were obtained from the study previously described [@garg2020survey].  The datasets were downloaded from [Gene Expression Omnibus (GEO)](https://www.ncbi.nlm.nih.gov/geo/).  We accessed DNA methylation data from a total 1, 417 individuals from [GSE51032](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE51032) and 
[GSE111629](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi) cohorts. The DNA methylation profiles were generated using the Illumina 450k Human Methylation BeadChip.  

The GSE51032 analysed primary cancers samples:  424 cancer free, 235 primary breast cancer, 166 
primary colorectal cancer and 20 other primary cancers. The GSE111629 cohort 335 Parkinson’s disease and 237 control samples.  

## Validation

We evaluated the performance of the method using True Positive Rate (TPR), False Positive Rate (FPR) and accuracy. We used the TPR to measure the proportion of detected epivariations by the `epimutations` approach  present in the validated (table \ref{tab:table_validated}) by [@garg2020survey]. FPR to calculate the identified epimutations outside the once found in [@garg2020survey], whether validated or not. The accuracy measures the closeness of the detected epimutation to the validated regions. 


We select samples differently depending on the study group and measure to compute. Control samples were selected randomly using different sample size: 20, 30, 40, 50, 60, 70, 80, 90 and 100. However, case samples were selected considering validated epimutations (for TPR and accuracy) or excluding epivariations found (for FPR) [@garg2020survey].  


The validated epimutations in table \ref{tab:table_validated} were only present on 5 individuals: GSM1235784 from GSE51032 cohort and GSM3035933, GSM3035791, GSM3035807 and GSM3035685 from GSE111629. Therefore, they were established as case samples when computing TPR and accuracy. Nevertheless, we compute FPR excluding the samples containing at least one epimutation found by [@garg2020survey].  For the remaining case samples, 4 were selected randomly in each execution.

We executed 100 times the same process for each control sample size. We defined for the analysis regions of $\approx$ 20 kb containing $\ge$ 3 GpGs. 

```{r tableEpiValidated, echo=FALSE, tab.cap='(ref:tableEpiValidated)'}
load(file.path(getwd(),"hpc/inputs/epi_validated.rda"))
epi_validated <- as.data.frame(epi_validated)
epi_validated$samples <- c("GSM1235784/GSM3035791",
                           "GSM3035685",
                           "GSM3035933", 
                           "GSM3035791/GSM3035807")
colnames(epi_validated) <- c("Chromosome", "Start", "End", 	"Width", "Strand", "Samples")
knitr::kable(epi_validated, format = "markdown", caption = "validated epimutations (Garg et al. 2020).\\label{tab:table_validated}")
```

Additionally, we have plotted the methylation values of the samples in the regions where the validated epimutations were found.
 
```{r graph_GSM1235784, echo = FALSE, fig.cap = "GSE51032 cohort samples in the region chr17:46018654-46019184", fig.align = 'center', fig.pos = "H"}
knitr::include_graphics(file.path(getwd(),"vignette/fig/GSM1235784_chr17-46018654-46019184.png"))
``` 

```{r graph_GSM3035685, echo = FALSE, fig.cap = "GSE111629 cohort samples in the region chr19:11199851-11200146", fig.align = 'center', fig.pos = "H"}
knitr::include_graphics(file.path(getwd(),"vignette/fig/GSM3035685_chr19-11199851-11200146.png"))
``` 

```{r graph_GSM3035791, echo = FALSE, fig.cap = "GSE111629 cohort samples in the region chr5:67584194-67584380", fig.align = 'center', fig.pos = "H"}
knitr::include_graphics(file.path(getwd(),"vignette/fig/GSM3035791_chr5-67584194-67584380.png"))
``` 

```{r graph_GSM3035791.2, echo = FALSE, fig.cap = "GSE111629 cohort samples in the region chr17:46018654-46019184", fig.align = 'center', fig.pos = "H"}
knitr::include_graphics(file.path(getwd(),"vignette/fig/GSM3035791_chr17-46018654-46019184.png"))
``` 

```{r graph_GSM3035807, echo = FALSE, fig.cap = "GSE111629 cohort samples in the region chr5:67583972-67584380", fig.align = 'center', fig.pos = "H"}
knitr::include_graphics(file.path(getwd(),"vignette/fig/GSM3035807_chr5-67583972-67584380.png"))
``` 

```{r graph_GSM3035933, echo = FALSE, fig.cap = "GSE111629 cohort samples in the region chr5:10249761-10251252", fig.align = 'center', fig.pos = "H"}
knitr::include_graphics(file.path(getwd(),"vignette/fig/GSM3035933_chr5-10249761-10251252.png"))
``` 

## Results

We compared GSM1235784 case sample against randomly selected control samples from GSE51032. furthermore,  GSM3035933, GSM3035791, GSM3035807 and GSM3035685 case samples were studied against controls from GSE111629 specifying a region of 20 kb and $\ge$ 3 GpGs. 

We obtained similar results in both cohorts. We observed that the methods manova, mahalanobis distance, multivariate linear models, quantile and beta identified the validated epimutations with a TPR of $>99\%$ even if the control sample is small. However,  the TPR in isolation forest increases together with the number of control samples obtaining a TPR $\ge$ 75 with 50 control samples or more. Regarding the accuracy, all the statistical  approaches detect the epivariants with $>80\%$ of closeness to the validated epimutations. 

We detected possible epivariations outside the epimutations found by [@garg2020survey] selecting random regions of 20 kb and $\ge$ 3 GpGs for both, control and case samples. We compared individual methylation profiles of a single case 
sample against control samples. We observed that in both cohorts and for every approach the FPR value was minimum $<0.01\%$. 

```{r table_GSE51032, echo = FALSE}
library(dplyr)
load("result_files/3-TPR_FPR_accuracy/TPR_FPR_accuracy_GSE51032.rda")
df <- df$chr17_46018653_46019185
df[,3:5] <- apply(df[,c(-1, -2)], 2,as.numeric)
kableExtra::kbl(list(df[1:30,], df[31:54,])) %>%
  kableExtra::kable_paper("striped", full_width = F)
#kableExtra::kbl(df[,-2], align = 'c') %>%
#    kableExtra::kable_paper("striped", full_width = F) %>%
#    kableExtra::pack_rows(index = c("n20" = 6, "n30" = 6, "n40" = 6, 
#                          "n50" = 6, "n60" = 6, "n70" = 6, 
#                          "n80" = 6, "n90" = 6, "n100" = 6),
#              label_row_css = "background-color: #666; color: #fff;")
#knitr::kable(mean_table, align = "lccrr", caption = "`epimutations` function TPR, FPR and accuracy using GSE51032 cohort")
```



```{r graph_GSE51032, echo = FALSE, fig.cap="epimutations performance for GSE51032 cohort detecting the epivariation located in chr5:10249760-10251253", fig.align='center', fig.pos = "H"}
knitr::include_graphics(file.path(getwd(),"result_files/4-Graph/GSE51032_chr17_46018653_46019185.png"))
```

```{r table_GSE111629, echo = FALSE, results='asis'}
load("result_files/3-TPR_FPR_accuracy/TPR_FPR_accuracy_GSE111629.rda")
df <- as.data.frame(data.table::rbindlist(df))
df[,3:5] <- apply(df[,c(-1, -2)], 2,as.numeric)
mean_table <- aggregate( .~ method + n, data = df, FUN= "mean")
kableExtra::kbl(list(mean_table[1:30,], mean_table[31:54,])) %>%
  kableExtra::kable_paper("striped", full_width = F)
#kableExtra::kbl(mean_table, align = 'c') %>%
#    kableExtra::kable_paper("striped", full_width = F) %>%
#    kableExtra::pack_rows(index = c("n20" = 6, "n30" = 6, "n40" = 6, 
#                          "n50" = 6, "n60" = 6, "n70" = 6, 
#                          "n80" = 6, "n90" = 6, "n100" = 6),
#              label_row_css = "background-color: #666; color: #fff;")
#knitr::kable(mean_table, align = "lccrr", caption = "`epimutations` function TPR, FPR and accuracy for GSE111629 cohort. The measures shown are the mean for all the validated epimutations identified.")
```

```{r graph_GSE111629_1, echo = FALSE, fig.cap="epimutations performance using GSE111629 cohort to detect the epivariation located in chr5:10249760-10251253", fig.align='center', fig.pos = "H"}
knitr::include_graphics(file.path(getwd(),"result_files/4-Graph/GSE111629_chr5_10249760_10251253.png"))
```

```{r graph_GSE111629_2, echo = FALSE, fig.cap="epimutations performance using GSE111629 cohort to detect the epivariation located in chr5:67583971-67584381", fig.align='center'}
knitr::include_graphics(file.path(getwd(),"result_files/4-Graph/GSE111629_chr5_67583971_67584381.png"))
```

```{r graph_GSE111629_3, echo = FALSE, fig.cap="epimutations performance using GSE111629 cohort to detect the epivariation located in chr17:46018653-46019185", fig.align='center'}
knitr::include_graphics(file.path(getwd(),"result_files/4-Graph/GSE111629_chr17_46018653_46019185.png"))
```

```{r graph_GSE111629_4, echo = FALSE, fig.cap="epimutations performance using GSE111629 cohort to detect the epivariation located in chr5:11199850-11200147", fig.align='center'}
knitr::include_graphics(file.path(getwd(),"result_files/4-Graph/GSE111629_chr19_11199850_11200147.png"))
```


\newpage

# Acknowledgements

We acknowledge the organizers of the [European BioHackathon 2020](https://www.biohackathon-europe.org/) for their support.

All the team members of *Project #5* for the contribution to this package: 

| Name | Surname | ORCID | Affiliation | Team |
|:-----|:--------|:-----:|:------------|:-----|
| Leire | Abarrategui | 0000-0002-1175-038X | Faculty of Medical Sciences, Newcastle University, Newcastle-Upon-Tyne, UK; Autonomous University of Barcelona (UAB), Barcelona, Spain | Development |
| Lordstrong | Akano | 0000-0002-1404-0295 | College of Medicine, University of Ibadan | Development |
| James | Baye | 0000-0002-0078-3688 | Wellcome/MRC Cambridge Stem Cell Institute, University of Cambridge, Cambridge CB2 0AW, UK; Department of Physics, University of Cambridge, Cambridge CB2 3DY, UK | Development |
| Alejandro | Caceres | - | ISGlobal, Barcelona Institute for Global Health, Dr Aiguader 88, 08003 Barcelona, Spain; Centro de Investigación Biomédica en Red en Epidemiología y Salud Pública (CIBERESP), Madrid, Spain | Development |
| Carles | Hernandez-Ferrer | 0000-0002-8029-7160 | Centro Nacional de Análisis Genómico (CNAG-CRG), Center for Genomic, Regulation; Barcelona Institute of Science and Technology (BIST), Barcelona, Catalonia, Spain | Development |		
| Pavlo | Hrab | 0000-0002-0742-8478 | Department of Genetics and Biotechnology, Biology faculty, Ivan Franko National University of Lviv | Validation |
| Raquel | Manzano | 0000-0002-5124-8992 | Cancer Research UK Cambridge Institute; University of Cambridge, Cambridge, United Kingdom | Reporting |
| Margherita | Mutarelli | 0000-0002-2168-5059 | Institute of Applied Sciences and Intelligent Systems (ISASI-CNR) | Validation |
| Carlos | Ruiz-Arenas | 0000-0002-6014-3498 | Centro de Investigación Biomédica en Red de Enfermedades Raras (CIBERER), Barcelona, Spain; Universitat Pompeu Fabra (UPF), Barcelona, Spain | Reporting |

# References
