# Supplemental material of published manuscripts

This directory contains scripts and files supporting publications from the BRGE lab. These include:

**1. Ruiz-Arenas, Carlos et al., Genome Research, 2020** [(Supp Material)](https://github.com/isglobal-brge/Supplementary-Material/tree/master/Ruiz-Arenas_2020)

> [Ruiz-Arenas et al.](https://www.biorxiv.org/content/10.1101/792747v1), 2020. A method to identify chromosomal subpopulations based on their recombination histories advances the study of the genetic basis of phenotypic traits.  
> 
> Recombination is a main source of genetic variability. However, the potential role of the variation generated by recombination in phenotypic traits, including diseases, remains unexplored as there is currently no method to infer chromosomal subpopulations based on recombination patterns differences. We developed recombClust, a method that uses SNP-phased data to detect differences in historic recombination in a chromosome population. We validated our method by performing simulations and by using real data to accurately predict the alleles of well known recombination modifiers, including common inversions in Drosophila melanogaster and human, and the chromosomes under selective pressure at the lactase locus in humans. We then applied recombClust to the complex human 1q21.1 region, where nonallelic homologous recombination produces deleterious phenotypes. We discovered and validated the presence of two different recombination histories in these regions that significantly associated with the differential expression of ANKRD35 in whole blood and that were in high linkage with variants previously associated with hypertension. By detecting differences in historic recombination, our method opens a way to assess the influence of recombination variation in phenotypic traits.


**2. Abarrategui, Leire et al., 2021** [(Supp Material)](https://github.com/isglobal-brge/Supplementary-Material/tree/master/Abarrategui_2021)
>
> Rare diseases are pathologies with a low prevalence (< 1 per 2,000 people). Most of these pathologies have an onset during childhood and a strong genetic etiology. Due to their low prevalence, there is a lack of knowledge which causes a delay in diagnosis and a lack of effective treatment. Thus, affecting the life expectancy and quality of the patient. Current methodologies for identifying mutations related to rare diseases have relied on Whole Genomic Sequencing. Despite exhaustive assessments, in a large proportion of case subjects, the underlying genetic etiology is not identified or the clinical assessment does not indicate a diagnosis. In order to overcome this difficulty, genome-wide DNA methylation analysis has been proposed to facilitate the molecular diagnosis of unresolved clinical cases and be consider for routine clinical assessment.  We develop epimutacions, a method that detects outliers in DNA methylation patterns associated with the diseases as proposed by [Aref-Eshghi et al., 2019](https://www.sciencedirect.com/science/article/pii/S0002929719301041).  We validated our method by performing simulations based on the data and results obtained by [Garg et al., 2020](https://www.sciencedirect.com/science/article/abs/pii/S0002929720302883). 

**3. Pelegri-Siso, Dolors et al., 2021** [(Supp Material)](https://github.com/isglobal-brge/Supplementary-Material/tree/master/Pelegri-Siso_2021)
>

**4. Carreras-Gallo, Natalia et al., 2021** [(Supp Material)](https://github.com/isglobal-brge/Supplementary-Material/tree/master/Carreras-Gallo_Caceres_2021)

> The effect of polymorphic inversions on DNA methylation and its modulation by the early-life exposome 
> 
> Polymorphic genomic inversions are chromosomal variants with intrinsic variability that play important roles in evolution, environmental adaptation, and complex traits. We investigated the DNA methylation patterns of three common human inversions, at 8p23.1, 16p11.2, and 17q21.31 in 1,009 blood samples from children and in 39 prenatal heart tissue samples. We found inversion-state specific methylation patterns within and nearby flanking each inversion region in both datasets. Additionally, numerous inversion-exposure interactions on methylation levels were identified from early-life exposome data comprising 64 exposures. For instance, children homozygous at inv-8p23.1 and higher meat intake were more susceptible to TDH hypermethylation (P=3.8x10-22); being the inversion, exposure, and gene known risk factors for adult obesity. Inv-8p23.1 associated hypermethylation of GATA4 was also detected across numerous exposures. Our data indicate that the pleiotropic influence of inversions during development and lifetime is substantially mediated by allele-specific methylation patterns which can be modulated by the exposome. 



**5. Caceres, Alejandro and Gonzales, JUan R 2022** [(Supp Material)](https://github.com/isglobal-brge/Supplementary-Material/tree/master/Caceres_2022)

> teff: estimation of Treatment EFFects on transcriptomic data with casual random forest
> 
> Causal inference on high dimensional feature data can be used to find a profile of patients who will benefit the most from treatment rather than no treatment. However, there is a need for usable implementations for transcriptomic data. We devel-oped teff that applies random causal forest on gene expression data to target individuals with high expected treatment effects.

> We applied teff to extract a profile of high benefit of treating psoriasis with brodalumab and observed that it was associat-ed with high T cell abundance in non-lesional skin at baseline and a low response to etanercept in an independent study. Individual patient targeting with causal inference profiling can inform the choice between treatments before the intervention begins.

**6. Caceres, Alejandro, Perez-Jurado, Luis A. and Gonzales, JUan R 2022** [(Supp Material)](https://github.com/isglobal-brge/Supplementary-Material/tree/master/Caceres_2022)

> Sexual-dimorphism in immune cell abundance in blood can contribute to the sex differences in cancer incidence and mortality. However, little is known about the genes associated with high immune sexual-dimorphism and their potential predictive value for disease. Using machine learning methods, we identified and validated the presence of a subpopulation where women have higher T cell abundance in blood than men. Females in the subpopulation upregulate XIST and TSIX, drivers of chromosome X inactivation and escapee genes (DDX3X, KDM5C and RPS4X), while men downregulate the homologs (DDX3Y, KDM5D and RPS4Y1), predictors of chromosome Y downregulation. This subpopulation of high immune female-biased dimorphism associated with high cancer survival and low cancer risk in females. Part of the female advantage on cancer risk and survival can thus be explained by immune dimorphisms residing in a set of genes located in both sex chromosomes.


