# Supplemental material of published manuscripts

This directory contains scripts and files supporting publications from the BRGE lab.

**1. Ruiz-Arenas, Carlos et al., Genome Research, 2020** [(Supp Material)](https://github.com/isglobal-brge/Supplementary-Material/tree/master/Ruiz-Arenas_2020)

> [Ruiz-Arenas et al.](https://www.biorxiv.org/content/10.1101/792747v1), 2020. A method to identify chromosomal subpopulations based on their recombination histories advances the study of the genetic basis of phenotypic traits.  
> 
> Recombination is a main source of genetic variability. However, the potential role of the variation generated by recombination in phenotypic traits, including diseases, remains unexplored as there is currently no method to infer chromosomal subpopulations based on recombination patterns differences. We developed recombClust, a method that uses SNP-phased data to detect differences in historic recombination in a chromosome population. We validated our method by performing simulations and by using real data to accurately predict the alleles of well known recombination modifiers, including common inversions in Drosophila melanogaster and human, and the chromosomes under selective pressure at the lactase locus in humans. We then applied recombClust to the complex human 1q21.1 region, where nonallelic homologous recombination produces deleterious phenotypes. We discovered and validated the presence of two different recombination histories in these regions that significantly associated with the differential expression of ANKRD35 in whole blood and that were in high linkage with variants previously associated with hypertension. By detecting differences in historic recombination, our method opens a way to assess the influence of recombination variation in phenotypic traits.


