# Supplemental material of published manuscripts

This directory contains scripts and files supporting publications from the BRGE lab. These include:

**1. Ruiz-Arenas, Carlos et al., Genome Research, 2020** [(Supp Material)](https://github.com/isglobal-brge/Supplementary-Material/tree/master/Ruiz-Arenas_2020)

> [Ruiz-Arenas et al.](https://www.biorxiv.org/content/10.1101/792747v1), 2020. A method to identify chromosomal subpopulations based on their recombination histories advances the study of the genetic basis of phenotypic traits.  
> 
> Recombination is a main source of genetic variability. However, the potential role of the variation generated by recombination in phenotypic traits, including diseases, remains unexplored as there is currently no method to infer chromosomal subpopulations based on recombination patterns differences. We developed recombClust, a method that uses SNP-phased data to detect differences in historic recombination in a chromosome population. We validated our method by performing simulations and by using real data to accurately predict the alleles of well known recombination modifiers, including common inversions in Drosophila melanogaster and human, and the chromosomes under selective pressure at the lactase locus in humans. We then applied recombClust to the complex human 1q21.1 region, where nonallelic homologous recombination produces deleterious phenotypes. We discovered and validated the presence of two different recombination histories in these regions that significantly associated with the differential expression of ANKRD35 in whole blood and that were in high linkage with variants previously associated with hypertension. By detecting differences in historic recombination, our method opens a way to assess the influence of recombination variation in phenotypic traits.


**2. Abarrategui, Leire et al., 2021** [(Supp Material)](https://github.com/isglobal-brge/Supplementary-Material/tree/master/Abarrategui_2021)
>
> Rare diseases are pathologies with a low prevalence (< 1 per 2,000 people). Most of these pathologies have an onset during childhood and a strong genetic etiology. Due to their low prevalence, there is a lack of knowledge which causes a delay in diagnosis and a lack of effective treatment. Thus, affecting the life expectancy and quality of the patient. Current methodologies for identifying mutations related to rare diseases have relied on Whole Genomic Sequencing. Despite exhaustive assessments, in a large proportion of case subjects, the underlying genetic etiology is not identified or the clinical assessment does not indicate a diagnosis. In order to overcome this difficulty, genome-wide DNA methylation analysis has been proposed to facilitate the molecular diagnosis of unresolved clinical cases and be consider for routine clinical assessment.  We develop epimutacions, a method that detects outliers in DNA methylation patterns associated with the diseases as proposed by [Aref-Eshghi et al., 2019](https://www.sciencedirect.com/science/article/pii/S0002929719301041).  We validated our method by performing simulations based on the data and results obtained by [Garg et al., 2020](https://www.sciencedirect.com/science/article/abs/pii/S0002929720302883). 

**3. Pelegri-Siso, Dolors et al., 2021** [(Supp Material)](https://github.com/isglobal-brge/Supplementary-Material/tree/master/Pelegri-Siso_2021)
>

**4. Carreras-Gallo, Natalia et al., 2021** [(Supp Material)](https://github.com/isglobal-brge/Supplementary-Material/tree/master/Carreras-Gallo_2021)
>
> Polymorphic inversions are chromosomic subsystems characterized by their own genetic variability. They play important roles in evolution, environmental adaptation, and complex traits. We investigated the methylation patterns in the blood of 1,005 children within three common human inversions, at 8p23.1, 17q21.31, and 16p11.2, and found that each inversion state had a specific pattern. The methylation patterns extended the inversion region and were sharply defined by the inversion breakpoints. Differentially methylated genes by the inversion genotypes overlapped with those differentially expressed. Numerous inversion-exposure interactions on methylation levels were identified from early-life exposome data comprising 114 exposures. In particular, inv-8p23.1 homozygous were susceptible to GATA4 hypermethylation with increased exposure to cadmium (P=1.7x10-9), both known risk factors of congenital heart disease (CHD). We confirmed the hypermethylation of GATA4 in heart tissue in CHD. Methylation patterns can substantially contribute to the influence of inversions on phenotypic traits which can be modulated by the exposome.  

